core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/io/SdkLengthAwareInputStream.java

```diff
@@ -85,15 +85,15 @@ public int available() throws IOException {
     }
 
     @Override
-    public void mark(int readlimit) {
+    public synchronized void mark(int readlimit) {
```

Comment:
These changes cause findbugs failures. See https://github.com/aws/aws-sdk-java-v2/actions/runs/6500168283/job/17654999932?pr=4588#step:3:67245

```
Error:  Medium: Inconsistent synchronization of software.amazon.awssdk.http.auth.aws.internal.signer.io.SdkLengthAwareInputStream.length; locked 50% of time [software.amazon.awssdk.http.auth.aws.internal.signer.io.SdkLengthAwareInputStream, software.amazon.awssdk.http.auth.aws.internal.signer.io.SdkLengthAwareInputStream, software.amazon.awssdk.http.auth.aws.internal.signer.io.SdkLengthAwareInputStream, software.amazon.awssdk.http.auth.aws.internal.signer.io.SdkLengthAwareInputStream] Unsynchronized access at SdkLengthAwareInputStream.java:[line 60]Unsynchronized access at SdkLengthAwareInputStream.java:[line 46]Synchronized access at SdkLengthAwareInputStream.java:[line 92]Synchronized access at SdkLengthAwareInputStream.java:[line 98] IS2_INCONSISTENT_SYNC
```

Reply:
Undoing this change in https://github.com/aws/aws-sdk-java-v2/pull/4588/commits/7a8c647008342ec95b948eb9b846a51e5774db63